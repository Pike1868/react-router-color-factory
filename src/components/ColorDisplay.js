import React from "react";
import { Link, useParams } from "react-router-dom";

/**
 * ColorDisplay Component:
 *
 * Displays an individual color. The color's hex value is set as the background color of the page, and color's name is shown in a message to the user.
 * Includes a link to return to the main list of colors.
 *
 * Props:
 * colorsArr (array): Array of color objects
 *
 */

const ColorDisplay = ({ colorsArr }) => {
  const { color } = useParams();
  const colorObj = colorsArr.find((c) => c.name === color);

  return (
    <div
      style={{
        backgroundColor: colorObj ? colorObj.hex : "transparent",
        textAlign: "center",
        width: "100vw",
        height: "100vh",
        padding: "20px",
      }}
    >
      <p
        style={{
          color: "#fff",
          fontSize: "2rem",
          fontWeight: "600",
          margin: "0 0 50px",
          padding: "50px 0",
        }}
      >
        This is {colorObj.name}.
      </p>
      <div>
        <Link
          to="/colors"
          style={{ color: "#fff", fontSize: "1.5rem", fontWeight: "500" }}
        >
          Go Back
        </Link>
      </div>
    </div>
  );
};

export default ColorDisplay;
