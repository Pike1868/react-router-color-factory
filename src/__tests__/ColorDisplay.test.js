import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router-dom";
import ColorDisplay from "../components/ColorDisplay";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    color: "red",
  }),
}));

describe("ColorDisplay Component Tests", () => {
  const initialColors = [
    { name: "red", hex: "#ff0000" },
    { name: "green", hex: "#008000" },
    { name: "blue", hex: "#0000ff" },
  ];

  //Smoke Test
  test("renders ColorDisplay without crashing", () => {
    render(
      <MemoryRouter initialEntries={["/colors/red"]}>
        <ColorDisplay colorsArr={initialColors} />
      </MemoryRouter>
    );
    const colorText = screen.getByText("This is red.");
    expect(colorText).toBeInTheDocument();
  });

  //Snapshot Test
  test("ColorDisplay matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/colors/red"]}>
        <ColorDisplay colorsArr={initialColors} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
