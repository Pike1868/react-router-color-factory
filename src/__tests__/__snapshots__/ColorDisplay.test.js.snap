// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorDisplay Component Tests ColorDisplay matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: rgb(255, 0, 0); text-align: center; width: 100vw; height: 100vh; padding: 20px;"
  >
    <p
      style="color: rgb(255, 255, 255); font-size: 2rem; font-weight: 600; margin: 0px 0px 50px; padding: 50px 0px;"
    >
      This is red.
    </p>
    <div>
      <a
        href="/colors"
        style="color: rgb(255, 255, 255); font-size: 1.5rem; font-weight: 500;"
      >
        Go Back
      </a>
    </div>
  </div>
</DocumentFragment>
`;
